app/controllers/magic_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class MagicController < ApplicationController
+  skip_before_action :authenticate
+
+  def index
+    unless FeatureService.enabled?(:dfe_sign_in_fallback)
+      redirect_to sign_in_path
+    end
+  end
+
+  def sign_in_by_email
+    render_not_found and return unless FeatureService.enabled?(:dfe_sign_in_fallback)
+
+    user = User.find_by(email: params.dig(:user, :email).downcase.strip)
+    if user&.dfe_sign_in_uid.present?
```

Comment:
Is there a reason that we only allow them in if they've used Signin before?

Reply:
> magic links as a fallback for when DfE Sign in is down

Is there a reason for the inverse? To allow them in if they've never used Signin Before?